logstash-core/src/main/java/org/logstash/settings/BaseSetting.java

```diff
@@ -193,7 +198,16 @@ public void format(List<String> output) {
     }
 
     public List<Setting<T>> withDeprecatedAlias(String deprecatedAlias) {
-        return SettingWithDeprecatedAlias.wrap(this, deprecatedAlias);
+        return withDeprecatedAlias(deprecatedAlias, Collections.emptyMap());
+    }
+    public List<Setting<T>> withDeprecatedAlias(String deprecatedAlias, Map<RubySymbol, String> kwargs) {
```

Comment:
I would _really_ like to keep `Ruby*` types out of the pure-java classes if we can help it, and would be glad to help brainstorm a way of keeping the complexity of calling into the pure-java bits from ruby either in ruby code or in a helper that is called from the ruby code.

Reply:
I see your concern to avoid further gluing of java ruby. The current case is simple enough that I can pass obsoleted_version as string instead of ruby symbol for both java and ruby. See the [commit](https://github.com/elastic/logstash/pull/16538/commits/e97d77e8a012daae36fe367974ba67dc3be34ba6)
I think this is enough for now.